### ISSUE TITLE
Replacing Textures

### ISSUE THREAD

**Issue by nullhook** [SpecialKO/SpecialK#60]
I'm using a tool for Sims 2 that infers vertex and pixel shaders. In a specific scene, I can see a list of vertex shaders, and when I click on a vertex shader, I can see the textures it uses. I'm wondering if it's possible to replace these textures. Is it possible to infer the dimension of the texture so we can replace it with the exact same dimensions? The format of the texture is available, so it should be easy to convert the new textures to that format.

<img width="375" alt="image" src="https://user-images.githubusercontent.com/8665427/233441632-c836e355-90eb-4f36-9af8-1204d6667a54.png">

**Comment by Aemony** [comment]
Texture modding in D3D9 is sorta in a questionable state since Microsoft made it impossible to debug that API on Windows 8.x and newer OSes. It's unlikely that any change related to it is going to be implemented.

**Comment by mirh** [comment]
If only someone could improve renderdoc's d3d9 driver..

**Comment by nullhook** [comment]
@mirh How would that help?

**Comment by mirh** [comment]
Sorry for the brainfart. 
Did somebody ever test the Windows [checked](https://stackoverflow.com/questions/29944572/d3d9-64-bit-debug-runtime#comment50018907_29944573) builds though?

**Comment by Aemony** [comment]
> Sorry for the brainfart.
> Did somebody ever test the Windows [checked](https://stackoverflow.com/questions/29944572/d3d9-64-bit-debug-runtime#comment50018907_29944573) builds though?

Yes, but they are too slow and unstable to be of much use. To give you an idea of how slow they are, just installing the OS takes like 1-2 hours, and booting it takes 10-15 minutes and ends in a BSOD.

Debugging is actually easier (*much easier*) done through a regular virtual Win7 machine with Visual Studio 2022 Remote Debugger installed on it. My use of that is what allowed me to fix Windows 7 support for both SKIF and Special K recently, as development occurs as usual on the main computer while remote debugging occurs from the host machine to the VM where the debug layer is enabled.

All that said, Windows 7 remains not officially supported and it is up to Kal if he wants to have a go at trying to set up this sort of debugging setup to improve on SK’s D3D9 support.

**Comment by Aemony** [comment]
The checked build is a no go entirely due to its huge slowdowns -- it's basically impossible to do anything for it.

Microsoft also doesn't provide checked builds for newer versions of Windows 10 so even mixing-matching DLL files are sure to cause issues due to those being more than 5 years out of date.

A Windows 7 VM is really the *only* way one can realistically debug D3D9 at this point.
